test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -106,6 +109,19 @@ func (r *DockerMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, err
 	}
 
+	dockerMachinePool.Status.InfrastructureMachineSelector = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			clusterv1.ClusterNameLabel:            cluster.Name,
+			clusterv1.MachinePoolOwnedLabel:       "true",
+			infraexpv1.DockerMachinePoolNameLabel: dockerMachinePool.Name,
+		},
+	}
+	dockerMachinePool.Status.InfrastructureMachineKind = "DockerMachine"
+	// Patch now so that the status and selectors are available.
```

Comment:
It's so that the MachinePool can see that the DockerMachinePool does support MPMs as soon as possible -- otherwise the MachinePool will spend more time reconciling fallback machines.

Reply:
I'm not sure this provides real benefits and in fact we are handling the racy condition in the MachinePool.
this requires a better solution, even if I fear the root of the issue is what is being discussed in https://github.com/kubernetes-sigs/cluster-api/pull/7938#discussion_r1154842474